discovery/marathon/marathon.go

```diff
@@ -298,15 +315,26 @@ func targetsForApp(app *App) []model.LabelSet {
 		if len(t.Ports) == 0 {
 			continue
 		}
-		target := targetForTask(&t)
-		targets = append(targets, model.LabelSet{
-			model.AddressLabel: model.LabelValue(target),
-			taskLabel:          model.LabelValue(t.ID),
-		})
+		for i := 0; i < len(t.Ports); i++ {
+			targetAddress := targetForTask(&t, i)
+			target := model.LabelSet{
+				model.AddressLabel: model.LabelValue(targetAddress),
+				taskLabel:          model.LabelValue(t.ID),
+			}
+			for ln, lv := range app.PortDefinitions[i].Labels {
+				ln = portDefinitionsLabelPrefix + ln
```

Comment:
I think it's necessary to sanitize the `ln` part first here. There should be an example in other discovery packages (hopefully). IIRC we have a special util package with such a method.